### PR TITLE
React: Add story-utils entry point

### PR DESCRIPTION
**Pull request by shilman**
Closes #22485

## What I did

Much simpler alternative to #22485, which is a potentially destabilizing change (unit tests are failing & it won't actually work without some changes to the virtual modules code per @tmeasday 's comment)

## How to test

In node this will not error:

```
const { composeStories } = require('@storybook/react/story-utils');
```

Whereas this will (due to some SSv6 code that assumes it's running in a browser);

```
const { composeStories } = require('@storybook/react');
```

We should fix this properly in 8.0, or simply delete SSv6 which should also fix the problem.

### 🦋 Canary release

This pull request has been released as version [`0.0.0-pr-22499-sha-edffc853`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-22499-sha-edffc853). Install it by pinning all your Storybook dependencies to that version.
<details>
<summary>More information</summary>

| | |
| --- | --- |
| **Published version** | [`0.0.0-pr-22499-sha-edffc853`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-22499-sha-edffc853) |
| **Triggered by** | @shilman |
| **Repository** | [storybookjs/storybook](https://github.com/storybookjs/storybook) |
| **Branch** | [`shilman/add-react-testing-api-entry`](https://github.com/storybookjs/storybook/tree/shilman/add-react-testing-api-entry) |
| **Commit** | [`edffc853`](https://github.com/storybookjs/storybook/commit/edffc853419f8f50235d3dd6852c517bea8c4eb6) |
| **Datetime** | Tue Nov 14 09:45:51 UTC 2023 (`1699955151`) |
| **Workflow run** | [6861840579](https://github.com/storybookjs/storybook/actions/runs/6861840579) |

To request a new release of this pull request, mention the `@storybookjs/core` team.

_core team members can create a new canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=22499`_
</details>

### CODE REVIEW

File: code/renderers/react/package.json [hunk 1]
```diff
@@ -31,6 +31,11 @@
       "require": "./dist/config.js",
       "import": "./dist/config.mjs"
     },
+    "./testing-api": {
```
Comment (yannbf) [inline code/renderers/react/package.json:34]: maybe we call it test-utils, like vue does with `@vue/test-utils`?
come to think of it, we'll be using `composeStories` for other 'portable stories' scenarios, right? not just testing. It's actually the only way to provide users a way to [reuse stories within stories](https://storybook.js.org/docs/react/writing-stories/stories-for-multiple-components) using CSF3, for instance.

Comment (jonniebigodes) [reply code/renderers/react/package.json:34]: Agree with it 👍 . Seems like a reasonable choice for what the feature aims to accomplish

Comment (shilman) [reply code/renderers/react/package.json]: Updated per feedback but I think we can leave this until we actually need it. When we delete SSv6 in v8 we won't actually need it.

**Comment by ndelangen** [comment]
@shilman considering we're dropping SSv6 suport entirely, I'll close this PR.
Feel free to re-open if you wish.

**Pull request closed without merging**